### DISCUSSION TITLE
Support for SAMD51 - Adafruit ItsyBitsy M4

### DISCUSSION THREAD

**Discussion by DJ-DotSoft** [Arduino-IRremote/Arduino-IRremote#1269] [Ideas]
The ReadMe specifically says SAMD51 is not supported, but I noticed there is reference to SAMD51 support in the changelog for v4.1.0 and v4.1.1. The changelog for v4.1.1 says "- SAMD51 use timer3 if timer5 not available.",  so I thought I would try with an Adafruit ItsyBitsy M4 (which doesn't have timer5). The code did not compile, but I made the following adjustments to IRTimer.hpp:

line 1584:
```
#define IR_SAMD_TIMER       TC3
#define IR_SAMD_TIMER_IRQ   TC3_IRQn
```
line 1620
```
    //GCLK->PCHCTRL[TC5_GCLK_ID].reg = GCLK_PCHCTRL_GEN_GCLK0_Val | (1 << GCLK_PCHCTRL_CHEN_Pos);
```
This compiles and works for my application (sending, not receiving).

So just thought I would post this and maybe someone who understands SAMD timers can verify if this is a good fix or not. Note that I can't see any changes in the code which would relate to the changelog comment, which may well indicate I haven't understood something!

**Comment by DJ-DotSoft** [comment]
An update already. It looks like the timers are only used for receiving (which I am not doing), so the changes above mean I can build the code, but it won't receive IR codes.

I think line 1620 can be changed to:

``
    GCLK->PCHCTRL[TC3_GCLK_ID].reg = GCLK_PCHCTRL_GEN_GCLK0_Val | (1 << GCLK_PCHCTRL_CHEN_Pos);
``
That at least might work, but I have not tested it.

**Reply by ArminJo** [reply]
Thanks for contributing! 👍 
Can you please test the latest version of this library fom this repo, I have included an automatic selection of timer3 if timer5 is not available.

**Reply by DJ-DotSoft** [reply]
Wow that was quick!

The code doesn't compile with the following error:

```
multiple definition of `TC3_Handler'; C:\Users\DomJ\AppData\Local\Temp\arduino\sketches\7B9D133378662EA72DCE7A1EA3F2296A\sketch\AmplifierPowerControl.ino.cpp.o:c:\Users\DomJ\Documents\Arduino\libraries\IRremote\src/private/IRTimer.hpp:1695: first defined here
collect2.exe: error: ld returned 1 exit status
```

But this is because I am using timer 3 for the Adafruit ZeroTimer library, and nothing to do with IRremote.

It compiles and works (for transmit) if I do this:

```
#define DISABLE_CODE_FOR_RECEIVER
#include <IRremote.hpp>
```

Thanks for the quick response.
